src/data/providers/subgraphPoolProvider.ts

```diff
@@ -166,6 +166,16 @@ export class SubgraphPoolProvider implements PoolDataProvider {
             pools = pools.filter((pool) => this.poolMatchesFilter(pool));
         }
 
+        // Until a full graft finishes on polygon/arbitrum, there are instances where the index is not
+        // present. For these instances, we default to the index of the token in the array.
```

Comment:
Is it deterministic that subgraph returns the pool tokens in the order expected here for all pool types?